src/validation/index.ts

```diff
@@ -135,9 +135,18 @@ export const validatorMiddleware: AdapterMiddlewareBuilder =
 
 export const errorCatchingMiddleware = (err: Error, req: FastifyRequest, res: FastifyReply) => {
   // Add adapter or generic error to request meta for metrics use
-  // There's a chance we have no request context if there was an error during input validation
+  // There's a chance we have no request context if there was an error during input validation,
+  // but we still want to include error in meta for metrics
   if (req.requestContext) {
     req.requestContext.meta = { ...req.requestContext?.meta, error: err }
+  } else {
+    req.requestContext = {
+      cacheKey: '',
+      data: undefined,
+      endpointName: '',
+      transportName: '',
```

Comment:
Let's make sure this won't increase cardinality on labels, meaning if the default for cacheKey is '' or undefined, same for endpointName and transportName